pkg/controller/worker/machines.go

```diff
@@ -48,10 +53,61 @@ func (w *workerDelegate) DeployMachineClasses(ctx context.Context) error {
 			return err
 		}
 	}
-	return w.seedChartApplier.Apply(
+
+	if err := w.seedChartApplier.Apply(
 		ctx, filepath.Join(kubevirt.InternalChartsPath, "machine-class"), w.worker.Namespace, "machine-class",
 		kubernetes.Values(map[string]interface{}{"machineClasses": w.machineClasses}),
-	)
+	); err != nil {
+		return fmt.Errorf("failed to apply machine-class chart: %v", err)
+	}
+
+	dataVolumeLabels := map[string]string{
+		clusterLabel: w.worker.Namespace,
+	}
+
+	kubeconfig, err := kubevirt.GetKubeConfig(ctx, w.Client(), w.worker.Spec.SecretRef)
+	if err != nil {
+		return fmt.Errorf("failed to get the kubeconfig of the kubevirt cluster: %v", err)
+	}
+
+	for _, machineClass := range w.machineClasses {
+		var (
+			storageClassName = pointer.StringPtr(machineClass["storageClassName"].(string))
+			pvcSize          = v1.ResourceList{v1.ResourceStorage: machineClass["pvcSize"].(resource.Quantity)}
+			sourceUrl        = machineClass["sourceURL"].(string)
+			machineClassName = machineClass["name"].(string)
+		)
+
+		dataVolumeSpec := cdi.DataVolumeSpec{
+			PVC: &v1.PersistentVolumeClaimSpec{
+				StorageClassName: storageClassName,
+				AccessModes: []v1.PersistentVolumeAccessMode{
+					"ReadWriteOnce",
+				},
+				Resources: v1.ResourceRequirements{
+					Requests: pvcSize,
+				},
+			},
+			Source: cdi.DataVolumeSource{
+				HTTP: &cdi.DataVolumeSourceHTTP{
+					URL: sourceUrl,
+				},
+			},
+		}
+
+		dataVolume := &cdi.DataVolume{}
+		dataVolume.Name = machineClassName
+		dataVolume.Spec = dataVolumeSpec
+		dataVolume.Labels = dataVolumeLabels
+
+		if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
You can't fail with `Conflict` if you are creating something. On the other hand, I think you should do `CreateOrUpdate` here, since the datavolume might already exist, and actually **will** exist in most case, since the worker is reconciled multiple times during the lifetime of a shoot. So what I would suggest:
* Rename the `KubevirtDataVolumeManager` method to `CreateOrUpdateVolume`.
* Use `controllerutil.CreateOrUpdate` inside this method to create or update the volume.